torchmultimodal/modules/losses/albef.py

```diff
@@ -0,0 +1,216 @@
+        ).mean()
+        loss_t2i = -torch.sum(
+            F.log_softmax(sim_t2i, dim=1) * sim_t2i_targets, dim=1
+        ).mean()
+
+        loss_itc = (loss_i2t + loss_t2i) / 2
+        return loss_itc
+
+
+class ImageTextMatchingLoss(nn.Module):
+    """
+    Compute the image-text matching loss by predicting whether an image-text pair is matched or not.
+
+    Args:
+        hidden_size (int): The image-text multimodal embedding hidden size.
+
+    Inputs:
+        embeddings_pos (Tensor of shape (batch_size, seq_length, hidden_size)):
+            The multimodal embeddings for positive image-text pairs.
+        embeddings_neg (Tensor of shape (2 * batch_size, seq_length, hidden_size)):
+            The multimodal embeddings for negative image-text pairs.
+    """
+
+    def __init__(
+        self,
+        hidden_size: int = 768,
+    ) -> None:
+        super().__init__()
+        self.itm_head = nn.Linear(
+            hidden_size, 2
+        )  # binary output indicating image-text matches
+
+    def forward(
+        self,
+        embeddings_pos: Tensor,
+        embeddings_neg: Tensor,
+    ) -> Tensor:
+        batch_size = embeddings_pos.size(0)
+        vl_embeddings = torch.cat(
+            [embeddings_pos[:, 0, :], embeddings_neg[:, 0, :]], dim=0
+        )
+        vl_output = self.itm_head(vl_embeddings)
+        itm_labels = torch.cat(
+            [
+                torch.ones(batch_size, dtype=torch.long),
+                torch.zeros(2 * batch_size, dtype=torch.long),
+            ],
+            dim=0,
+        ).to(vl_embeddings.device)
+        loss_itm = F.cross_entropy(vl_output, itm_labels)
+        return loss_itm
+
+
+class MaskedLanguageModelingLoss(nn.Module):
+    """
+    Compute the autoregressive masked language modeling loss by predicting the next token.
+
+    Args:
+        masked_token_id (int): The token id indicating a masked token. Default is -100.
+        vocab_size (int): The number of different tokens the prediction_head can predict. Default is 30522.
+        hidden_size (int): The hidden size of the prediction_head. Default is 768.
+        layer_norm_eps (float): The epsilon used by the prediction_head normalization layer. Default is 1e-12.
+        alpha (float): The interpolation value between mlm_loss and loss_distill. Default is 0.
+        transform_act_fn (Callable[[Tensor], Tensor]): The activation function in the prediction_head. Default is GELU.
+
+    Inputs:
+        labels (Tensor): The masked output tokens.
+        hidden_states (Tensor): The hidden states of preceding tokens.
+        hidden_states_m (Tensor): The hidden states of preceding tokens from momentum models.
+        weights (Tensor): The weight for each output.
+    """
+
+    def __init__(
+        self,
+        mask_token_id: int = -100,
+        vocab_size: int = 30522,
+        hidden_size: int = 768,
+        layer_norm_eps: float = 1e-12,
+        alpha: float = 0.0,
```

Comment:
Just curious about the choice of 0.0 as the default value here. I know they use it as the default in xbert.py, but I think in the [VQA config](https://github.com/salesforce/ALBEF/blob/main/configs/VQA.yaml#L16) they have it set as 0.4. (I actually like the choice of 0.0 as it means we are dealing with "true" MLM loss with no distillation, but just want to call out that it won't match the default model values.)